spec/intergration/user_show_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'rails_helper'
+
+RSpec.describe 'User Show', type: :feature do
+  describe 'User' do
+    before(:each) do
+      @photo = 'avatar.png'
+      @user1 = User.create! name: 'grayce', photo: @photo, password: '123456789', email: 'grayce@yahoo.com',
+                            bio: 'test bio', confirmed_at: Time.now
+      visit('/users/sign_in')
+      fill_in 'Email', with: 'grayce@yahoo.com'
+      fill_in 'Password', with: '123456789'
+      click_button 'Log in'
+
+      @post1 = Post.create!(author: @user1, title: 'test title 1', text: 'test text 1', likes_counter: '0',
+                            comments_counter: '0')
+      @post2 = Post.create!(author: @user1, title: 'test title 2', text: 'test text 2', likes_counter: '0',
+                            comments_counter: '0')
+      @post3 = Post.create!(author: @user1, title: 'test title 3', text: 'test text 3', likes_counter: '0',
+                            comments_counter: '0')
+      click_on 'grayce'
+    end
+
+    it 'shows the username of the user' do
+      expect(page).to have_content('grayce')
+    end
+
+    it 'shows the photo for ther user' do
+      image = page.all('img')
+      expect(image.size).to eql(0)
+    end
+
+    it 'shows number of posts for the user' do
+      expect(page).to have_content('Number of posts:')
+    end
+
+    it 'shows the bio for the user' do
+      expect(page).to have_content('test bio')
+      visit user_session_path
+    end
+
+    it 'Should see the first 3 posts for the user.' do
+      expect(page).to have_content 'test title 1'
+      expect(page).to have_content 'test title 2'
+      expect(page).to have_content 'test title 3'
+    end
+
+    it 'Can see a button that lets me view all of a users posts' do
+      expect(page).to have_button('See all posts')
+    end
+
+    it 'When I click to see all posts, it redirects me to the user\'s post\'s show page.' do
+      click_link 'test title 1'
+      expect(page).to have_current_path user_post_path(@user1, @post1)
+    end
```

Comment:
- [ ] This test is also failing. It's failing because currently, the content is not wrapped in `link`, instead `whole div` is wrapped`. Please change your code accordingly and wrap the content with `link` instead of the whole div. 

![Screenshot from 2022-03-11 14-56-32](https://user-images.githubusercontent.com/22434660/157839682-500306cf-cde1-424a-a7d4-7d71dc8d12ea.png)

Reply:
Thank you, corrected and the test is now passing: https://share.getcloudapp.com/12ug6LR4/annotate